Add docs structure to the repo and place for templates recipes (#224)
asyncapi/generator: docs/authoring.md

1. Templates may contain "filters" or helper functions. They must be stored in the `.filters` directory under the template directory. [Read more about filters](#filters).
1. Templates may have a configuration file. It must be stored in the template directory and its name must be `.tp-config.json`. [Read more about the configuration file](#configuration-file).
1. There are params with special meaning. [Read more about special params](#special-params).

> 1. Default parameters that you can use in your templates, except the ones that you defined on your own in the config,  are:

Reviewer: ```suggestion
1. The default variables you have access to in any the template file are the following:
```
Think we have to use `variable` instead of parameters, not to confuse it with the configuration parameters.

   - `asyncapi` that is a parsed spec file object. Read the [API](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument) of the Parser to understand to what structure you have access in this parameter.
   - `originalAsyncAPI` that is an original spec file before it is parsed. 

> 

Reviewer: ```suggestion
1. When using [file templating](https://github.com/asyncapi/generator#file-templating) you have access to the following extra variables:
   - When using `$$channel$$` you have access to two variables `channel` and `channelName`. Where `channel` contains the current channel being rendered.
   - When using `$$message$$` you have access to two variables `message` and `messageName`. Where `message` contains the current message being rendered.
   - When using `$$schema$$` you have access to two variables `schema` and `schemaName`. Where `schema` contains the current schema being rendered.
   - When using `$$parameter$$` you have access to two variables `parameter` and `parameterName`. Where `parameter` contains the current parameter being rendered.
   - When using `$$securityScheme$$` you have access to two variables `securityScheme` and `securitySchemeName`. Where `securityScheme` contains the current security scheme being rendered.
```
Think we should add the file template variables we have access to when using that feature 😄

|Name|Description|
|---|---|

> |`server`| It is used to let the template know which server we want to use. In some cases, this may be required. For instance, when generating code that connects to a specific server. If your template need your users to specify a server, use this param.

Reviewer: ```suggestion
|`server`| It is used to let the template know which server we want to use. In some cases, this may be required. For instance, when generating code that connects to a specific server. If your template need your users to specify a server, use this parameter.
```
